src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -399,6 +399,28 @@ class ConcatExpression : public detail::VariadicExpression {
 using EncodeForUriExpression =
     StringExpressionImpl<1, decltype(encodeForUriImpl)>;
 
+// TO_NUMERIC (int or double)
+auto InvalidArg = std::errc::invalid_argument;
+auto OutOfRange = std::errc::result_out_of_range;
+
+template <typename T>
+[[maybe_used]] auto toNumeric = [] (std::optional<std::string> input) {
+  if (!input.has_value()) {
+    return Id::makeUndefined();
+  } else {
+    T res{};
+    auto str = boost::lexical_cast<std::string>(*input);
+    auto conv = std::from_chars(str.data(), str.data() + str.size(), res);
```

Comment:
You probably have to use `absl::from_chars` from header `absl/strings/charconv` (we also have other usages in the code base). It is used in exactly the same way, but the floating point `std::from_chars` is not yet implemented on all platforms that we support.